### PR TITLE
Normalize use of backticks compiler messages

### LINKED ISSUE #60532: Normalize use of backticks in syntax for user-facing compiler messages
This is more pedantry from me, and admittedly not very high-priority, but I have noticed that there are discrepancies between compiler error/informational messages where syntax is included in them – some use backticks around code syntax, some don't. Furthermore, some start with lower case (the vast majority), while some don't.

For example:

```
"attribute must be of the form `#[derive(Trait1, Trait2, ...)]`"
```
vs.
```
"type does not implement `fmt::Debug`; consider adding #[derive(Debug)] \
```
(an internal inconsistency in this second case too!)

It would be nice to normalise this according  some guidelines. I'd imagine the guidelines should agree with rustdoc as much as possible, though starting all messages with lower case may be desirable, since it is "command-line fashion".

### PR DESCRIPTION
**Pull request by fakenine**
Normalize use of backticks compiler messages

Fixes https://github.com/rust-lang/rust/issues/60532

### CODE REVIEW

File: src/librustc/hir/lowering.rs [hunk 1]
```diff
@@ -1664,7 +1664,7 @@ impl<'a> LoweringContext<'a> {
                         if pos == ImplTraitPosition::Binding &&
                             nightly_options::is_nightly_build() {
                             help!(err,
-                                  "add #![feature(impl_trait_in_bindings)] to the crate attributes \
+                                  "add `#![feature(impl_trait_in_bindings)]` to the crate attributes \
```
Comment (Centril) [inline src/librustc/hir/lowering.rs]: This line is too long.

Comment (alexreg) [reply src/librustc/hir/lowering.rs]: Yeah. The cap is 100 chars. You can run `./x.py test src/tools/tidy` locally to check this.

File: src/librustc_passes/rvalue_promotion.rs [hunk 2]
```diff
@@ -338,14 +338,14 @@ fn check_expr_kind<'a, 'tcx>(
                     if v.in_static {
                         for attr in &v.tcx.get_attrs(did)[..] {
                             if attr.check_name(sym::thread_local) {
-                                debug!("Reference to Static(id={:?}) is unpromotable \
-                                       due to a #[thread_local] attribute", did);
+                                debug!("Reference to `Static(id={:?})` is unpromotable \
```
Comment (alexreg) [inline src/librustc_passes/rvalue_promotion.rs]: Maybe change all these and similar `debug!` expressions to begin with lower-case, as a matter of consistency/convention... not too important though. (And yes, this is pre-existing.)

**Review by alexreg — COMMENTED** [review]
Looks good, apart from the two very minor comments. Thanks a lot for this!

**Comment by alexreg** [comment]
I haven't checked it myself yet, but can you make sure that literal values are all quoted too? i.e., strings, numbers, bools, so you have (e.g.) `` `true` ``.

Actually, this can look kind of awkward with strings, e.g., `` `"foo"` ``. What do you think @Centril ?

File: src/librustc/hir/lowering.rs (hunk 1 above)
Comment (fakenine) [reply src/librustc/hir/lowering.rs]: My apologizes. That's noted! I will check this

**Comment by Centril** [comment]
> Actually, this can look kind of awkward with strings, e.g., `` `"foo"` ``. Even though it would be most consistent. What do you think @Centril?

Seems fine to me.

**Comment by fakenine** [comment]
Hello @alexreg! I have updated the PR. Unfortunately, I haven't been able to find debug messages with literal values that are not quoted...But maybe my regexes weren't accurate.

For this PR, I've squashed the 3 commits into one. Would you prefer 3 seperate commits for easier review ? (especially for the last big commit where the tests were update by the `x.py` command)

**Comment by alexreg** [comment]
@fakenine Thanks a lot. Yes, 3 separate commits would be best. I already reviewed the first one (presuming you didn't amend it), so that would save time.

As for literal values, probably this will have to be done manually I'm afraid. But the good news is there are probably many fewer examples of this. Maybe identify them from `.stderr`/`.stdout` files to get the full error messages, then search for them?

Looks like you forgot to bless the run-pass tests BTW. :-)

**Comment by alexreg** [comment]
@fakenine Looks like there's just a few more ui tests to bless... if you could do that, fix the little conflict (just rebase and bless the corresponding test), and make the small change I commented on, I'll review and hopefully we can get this merged. :-)

**Comment by fakenine** [comment]
Hello @alexreg! Thank you for the feedback :) I'll bless the remaining tests and apply the requested changes (blessing takes the most time because my computer takes long quite a long time to build hehe). I will notify you when everything is ready for me!

File: src/librustc/session/config.rs [hunk 3]
```diff
@@ -1401,7 +1401,7 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
     thinlto: Option<bool> = (None, parse_opt_bool, [TRACKED],
         "enable ThinLTO when possible"),
     inline_in_all_cgus: Option<bool> = (None, parse_opt_bool, [TRACKED],
-        "control whether #[inline] functions are in all cgus"),
+        "control whether `#[inline]` functions are in all cgus"),
```
Comment (alexreg) [inline src/librustc/session/config.rs]: `cgus` -> `CGUs` (and anywhere else)

File: src/librustc/session/config.rs [hunk 4]
```diff
@@ -1401,7 +1401,7 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
...
     tls_model: Option<String> = (None, parse_opt_string, [TRACKED],
         "choose the TLS model to use (rustc --print tls-models for details)"),
```
Comment (alexreg) [inline src/librustc/session/config.rs]: quotes around the command line: `` choose the TLS model to use (`rustc --print tls-models` for details) ``

If you could try grepping for command lines elsewhere (e.g., `rustc --`, `cargo --`, `rls --` or even something more flexible), that would be great.

File: src/librustc_codegen_utils/link.rs [hunk 5]
```diff
@@ -57,7 +57,7 @@ pub fn find_crate_name(sess: Option<&Session>,
         if let Some(ref s) = sess.opts.crate_name {
             if let Some((attr, name)) = attr_crate_name {
                 if name.as_str() != *s {
-                    let msg = format!("--crate-name and #[crate_name] are \
+                    let msg = format!("--crate-name and `#[crate_name]` are \
```
Comment (alexreg) [inline src/librustc_codegen_utils/link.rs]: `` `--crate-name` ``

Comment (alexreg) [reply src/librustc_codegen_utils/link.rs]: So, maybe grep for Rust string literals containing `--` or something?

File: src/libsyntax/feature_gate.rs [hunk 6]
```diff
@@ -2017,13 +2017,13 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
...
                     gate_feature_post!(&self, start, i.span,
-                                      "a #[start] function is an experimental \
+                                      "a `#[start]` function is an experimental \
                                        feature whose signature may change \
                                        over time");
                 }
                 if attr::contains_name(&i.attrs[..], sym::main) {
                     gate_feature_post!(&self, main, i.span,
-                                       "declaration of a nonstandard #[main] \
+                                       "declaration of a nonstandard `#[main]` \
```
Comment (alexreg) [inline src/libsyntax/feature_gate.rs]: `nonstandard` -> `non-standard` (maybe grep elsewhere too)

**Review by alexreg — CHANGES_REQUESTED** [review]
Overall very good.

File: src/libsyntax/parse/attr.rs [hunk 7]
```diff
@@ -226,7 +226,7 @@ impl<'a> Parser<'a> {
 
     fn parse_unsuffixed_lit(&mut self) -> PResult<'a, ast::Lit> {
         let lit = self.parse_lit()?;
-        debug!("Checking if {:?} is unusuffixed.", lit);
+        debug!("checking if {:?} is unusuffixed.", lit);
```
Comment (alexreg) [inline src/libsyntax/parse/attr.rs]: Remove `.`.

**Review by alexreg — CHANGES_REQUESTED** [review]

**Comment by alexreg** [comment]
Great, thanks. Just a few more things to address (pre-existing, so not your fault), and we're good to go. If you have trouble blessing all the tests `./x.py test --bless` should do it (make sure you run it a second time with `--compare-mode=nll`) – or it's simply taking you too long – let me know and I can give it a go.

As for building, yeah, it's a pain... make sure you do `./x.py your_cmd -i --keep-stage 0` to speed things up though, and maybe consider the [GCC build farm](https://internals.rust-lang.org/t/gcc-compile-farm-for-rustc/9511).

**Comment by fakenine** [comment]
Hello @alexreg! Thanks again for the advices :) I've applied the requested changes and ran the few missing `--bless` on test (`test/ui` and `test/rustdoc-ui`, normally). This should be good for me, unless the pipeline catches something I would have missed.

**Comment by alexreg** [comment]
@fakenine Great, thanks. Everything looks good to me too now. (By the way, feel free to mark comments as "resolved" on GitHub if you've addressed them, but I've done the existing ones for you.)

Unfortunately I only saw this just now, so there's a new conflict. Hopefully however you can just resolve the conflict by adjusting line numbers manually – simply comparing the line numbers in the error messages in `feature-gate-slice-patterns.stderr` and those of `feature-gate-slice-patterns.rs`. This will save you having to re-build and bless everything! Then just force-push (squash if you like). Ping me when you do this, and I'll approve this PR.

**Comment by fakenine** [comment]
@alexreg Thank you! I've resolved the little conflict.

**Comment by alexreg** [comment]
@fakenine Super. I appreciate you taking this on, and your patience throughout the revisions and "blessings"... hopefully this will merge without problems.

Now that you've been blooded to compiler development, feel free to look for another issue and tackle it if you fancy. If there are no mentoring instructions and you can't find someone to advise quickly, tag/ping me and I'll try to help.

**Comment by fakenine** [comment]
@alexreg Great. Thanks to you too for taking the time to review this PR. I will gladly look into other issues in this repo!

PS: I've resolved the other conflict that popped during the night

**Comment by fakenine** [comment]
Looks like I’ve missed something while resolving the conflict. My bad ! :/ I will look into this  tonight

**Comment by alexreg** [comment]
@fakenine Don't worry, I've already fixed it (I hope). Apparently it was @Centril's fault for sneaking in a new test. ;-)

**Comment by Centril** [comment]
cc @pietroalbini @Mark-Simulacrum Highfive is being silly.

**Comment by alexreg** [comment]
@fakenine Looks like the last problem was caused by you accidentally committing an submodule hash for `src/tools/rust-installer`. This is sadly all too easy to do when rebasing (done it a dozen times myself), but just something to note. :-)

(The solution is to remember to run `git submodule update --init --recursive` after every rebase (before any commit is made, even during the rebase). Maybe a local hook can be configured for this?

**Comment by fakenine** [comment]
@alexreg Oops! My bad, that's noted for next time. I'll fix this

**Comment by alexreg** [comment]
@fakenine I’d already fixed it myself before posting that comment.

**Comment by Centril** [comment]
Tests seem to have failed in the PR builder so @bors r- retry

Please consider breaking this PR up into several independent pieces.
It makes the chances of conflicts and problems with various other PRs much lower.

**Comment by alexreg** [comment]
@fakenine Looks like some recent merges created new tests that need blessing... sorry! https://dev.azure.com/rust-lang/rust/_build/results?buildId=2537

**Comment by alexreg** [comment]
@bors delegate=fakenine

You can now r+ it in a similar manner to how I did above, once you've reblessed the tests. (Make sure you pull my latest version from your branch before you edit it though.) Maybe ping @Centril once you've r+'ed too, in case I'm not around. He can make sure it gets through without further conflicts.. hopefully!

**Comment by Mark-Simulacrum** [comment]
Note: please do @ bors r=alexreg and not @ bors r+, since the latter will incorrectly approve this PR in your name, not alexreg's. The spaces here shouldn't be replicated, that's merely because bors is very eager in finding invocations to it.

**Comment by alexreg** [comment]
Yes sorry, @Mark-Simulacrum is right. It's very late here. -_-

**Comment by fakenine** [comment]
@Centril I’ll do that. I’m closing this PR and will split it to avoid multiple rebasings and conflicts

**Pull request closed without merging**